frontend/model/contracts/group.js

```diff
@@ -686,6 +695,15 @@ sbp('chelonia/defineContract', {
             })
           // TODO - #828 remove other group members contracts if applicable
         } else {
+          const myProfile = state.profiles[username] || null
+
+          if (!meta.isRemoveOurselves &&
+            isContractYoungerThanUser(meta, myProfile)) {
```

Comment:
Since, per the other comment I posted, you can do all the checks here in this function, you no longer need the `isContractYoungerThanUser` I don't think, and you should use `compareISOTimestamps` from `shared/time.js` instead.